x-pack/plugins/ecs_data_quality_dashboard/server/schemas/result.ts

```diff
@@ -39,5 +39,17 @@ export type ResultDocument = t.TypeOf<typeof ResultDocument>;
 
 export const PostResultBody = ResultDocument;
 
-export const GetResultParams = t.type({ pattern: t.string });
-export type GetResultParams = t.TypeOf<typeof GetResultParams>;
+export const GetResultsIndicesLatestParams = t.type({ pattern: t.string });
+export type GetResultsIndicesLatestParams = t.TypeOf<typeof GetResultsIndicesLatestParams>;
+
+export const GetResultsIndicesPatternParams = t.type({
+  pattern: t.string,
+});
+
+export const GetResultsIndicesPatternQuery = t.type({
+  limit: t.union([t.number, t.undefined]),
+  from: t.union([t.number, t.undefined]),
+  startDate: t.union([t.string, t.undefined]),
+  endDate: t.union([t.string, t.undefined]),
+  outcome: t.union([t.literal('pass'), t.literal('fail'), t.undefined]),
+});
```

Comment:
NIT: We can use `t.partial` so we don't have to do the union with `undefined` in each parameter:

```suggestion
export const GetResultsIndicesPatternQuery = t.partial({
  limit: t.number,
  from: t.number,
  startDate: t.string,
  endDate: t.string,
  outcome: t.union([t.literal('pass'), t.literal('fail')),
});
```

Reply:
thanks for catching, will change